mirage/views/osf-resource.ts

```diff
@@ -52,7 +52,15 @@ export function osfResource(
     }
 
     if (actions.includes('show')) {
-        server.get(detailPath, mirageModelName);
+        server.get(detailPath, function(schema, request) {
+            const models = schema[mirageModelName]
+                .where(m => filter(m, request))
```

Comment:
Looks like this just pulls the first model of `mirageModelName` every time. `schema[mirageModelName].find(request.params.id)` should give you the model you're looking for.